dao/src/main/java/org/thingsboard/server/dao/sql/JpaAbstractDao.java

```diff
@@ -51,6 +62,33 @@ protected void setSearchText(E entity) {
     @Override
     @Transactional
     public D save(TenantId tenantId, D domain) {
+        E entity = prepare(domain);
+        entity = getRepository().save(entity);
+        return DaoUtil.getData(entity);
+    }
+
+    @Override
+    @Transactional
+    public D saveAndFlush(TenantId tenantId, D domain) {
+        D d = save(tenantId, domain);
+        getRepository().flush();
+        return d;
+    }
+
+    @Override
+    public D create(TenantId tenantId, D domain) {
```

Comment:
looks kind of complicated. Is it not enough just `@Transactional`?

Reply:
It would require autowiring `self`, because `create` is called directly from the dao and `@Transactional` won't make a difference.